### PR TITLE
fix nil pointer error when using `rly config show` after using `rly paths new`

### PR DESCRIPTION
**Pull request by jtieri**
The `ChannelFilter` was not being initialized on the `Path` struct before being written to the config file on disk in `rly paths new` so then when it is parsed in `initConfig()` and `ValidateChannelFilterRule()` is called there is a nil pointer error when attempting to access the fields of the `ChannelFiler`

### CODE REVIEW

**Review by mark-rushakoff — COMMENTED** [review]
I think this approach probably causes more trouble than it solves. 

AFAICS, `ChannelFilter.Rule` is already allowed to be empty, and `ChannelFilter.ChannelList` is allowed to be nil or empty.

I would suggest changing Path's field `Filter *ChannelFilter` to `Filter ChannelFilter` so that the zero value is useful. `Path` values are already passed by reference anyway so there is not a clear advantage to using `*ChannelFilter` references, since those values are small and unlikely to be shared.

File: cmd/start.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ $ %s start demo-path2 --max-tx-size 10`, appName, appName)),
 			ctx, cancel := context.WithCancel(cmd.Context())
 			defer cancel()
 
-			errorChan := relayer.StartRelayer(ctx, c[src], c[dst], filter, maxTxSize, maxMsgLength)
+			errorChan := relayer.StartRelayer(ctx, c[src], c[dst], &filter, maxTxSize, maxMsgLength)
```
Comment (mark-rushakoff) [inline cmd/start.go]: I would switch the other uses away from pointers too at this point, but we can fix that separately.

**Review by mark-rushakoff — APPROVED** [review]

File: cmd/start.go (hunk 1 above)
Comment (jtieri) [reply cmd/start.go]: will circle back on this in another PR right now

**Comment by jtieri** [comment]
Failed CI so just went ahead and removed the pointers from the function parameters also

**Review by mark-rushakoff — APPROVED** [review]

**Pull request merged**